File: 1_matrices/matrix.ts
/*
Heru Handika
25 October 2020
Add identity matrix
*/

function create_identity_matrix(size: number): number[][]{
    let mat: any[] = [];
    for (let rows = 0; rows < size; rows++) {
        mat[rows] = [];
        for (let cols = 0; cols < size; cols++) {
            if (rows === cols) {
                mat[rows][cols] = 1;
            } else {
                mat[rows][cols] = 0;
            }
        }
    }
    return mat;
}

function print_matrix(vec: number[][]) {
    let rows = 0;
    while (rows < vec.length) {
        console.log(mat[rows] + '\n');
        rows++;
    }
}

let mat: number[][] = create_identity_matrix(5);
console.log("Identity matrix: \n");
print_matrix(mat);

File: 1_matrices/rust/src/matrix.rs
pub mod matrix {
    pub fn print_matrix(mat: &Vec<Vec<i32>>) {
        for i in mat.iter() {
            println!("{:?}", i);
        }
    }

    pub fn create_identity_matrix(row_size: usize, col_size: usize) -> Vec<Vec<i32>>{
        let mut mat = vec![Vec::with_capacity(row_size); col_size];
        for i in 0..mat.len() {
            for j in 0..mat.len() {
                if i == j {
                    mat[i].push(1);
                } else {
                    mat[i].push(0);
                }
            }
        }
       mat
    }
}



